[Materials] Add rendering support for blur effects

pxlcoder opened the pull request:
#### fc46a347192024cf8083fb284cd815e0e399eb2f
<pre>
[Materials] Add rendering support for blur effects
<a href="https://bugs.webkit.org/show_bug.cgi?id=284729">https://bugs.webkit.org/show_bug.cgi?id=284729</a>
<a href="https://rdar.apple.com/141527633">rdar://141527633</a>

Reviewed by Simon Fraser.

Create and display `MTMaterialLayer`s when using `-apple-visual-effect` with
one of the `-apple-system-blur-` keywords. Note that this is not web-exposed.

`MTMaterialLayer`s are backdrop layers with a specific set of `CAFilter`s,
depending on the blur effect.

* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-chrome-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-chrome.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thick-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thick.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thin-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thin.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-ultra-thin-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-ultra-thin.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material.html: Added.
* LayoutTests/apple-visual-effects/backdrop-filter-with-apple-visual-effect-expected.html: Added.
* LayoutTests/apple-visual-effects/backdrop-filter-with-apple-visual-effect.html: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-chrome-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-thick-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-thin-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-ultra-thin-expected.txt: Added.
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/PAL/pal/cocoa/CoreMaterialSoftLink.h: Copied from Source/WebCore/platform/cocoa/AppleVisualEffect.h.
* Source/WebCore/PAL/pal/cocoa/CoreMaterialSoftLink.mm: Copied from Source/WebCore/platform/cocoa/AppleVisualEffect.h.
* Source/WebCore/PAL/pal/spi/cocoa/CoreMaterialSPI.h: Copied from Source/WebCore/platform/cocoa/AppleVisualEffect.h.
* Source/WebCore/page/FrameSnapshotting.cpp:
(WebCore::styleContainsComplexBackground):
* Source/WebCore/platform/cocoa/AppleVisualEffect.cpp:
(WebCore::appleVisualEffectNeedsBackdrop):

Add a helper method to distinguish visual effects which create a backdrop (blur)
from those that do not (vibrancy). Support for vibrancy will be added in a
subsequent patch.

* Source/WebCore/platform/cocoa/AppleVisualEffect.h:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::needsBackdrop const):
(WebCore::GraphicsLayer::dumpProperties const):
* Source/WebCore/platform/graphics/GraphicsLayer.h:
(WebCore::GraphicsLayer::appleVisualEffect const):
(WebCore::GraphicsLayer::setAppleVisualEffect):

Specify `BackdropFiltersChanged` in addition to `AppleVisualEffectChanged`, as
material blurs set filters on a backdrop. A structural layer must be created,
and the backdrop filter rect must also be updated.

(WebCore::GraphicsLayer::needsBackdrop const): Deleted.
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::setAppleVisualEffect):
(WebCore::GraphicsLayerCA::commitLayerChangesBeforeSublayers):
(WebCore::GraphicsLayerCA::updateBackdropFilters):
(WebCore::GraphicsLayerCA::updateAppleVisualEffect):

Only support backdrop effects for now. Foreground effects (vibrancy) will be
implemented in a subsequent patch.

(WebCore::GraphicsLayerCA::purposeNameForInnerLayer const):
(WebCore::GraphicsLayerCA::layerChangeAsString):
(WebCore::GraphicsLayerCA::changeLayerTypeTo):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.h:
* Source/WebCore/platform/graphics/ca/PlatformCALayer.h:

Introduce a new layer type for materials.

* Source/WebCore/platform/graphics/ca/PlatformCALayer.mm:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/ca/cocoa/PlatformCALayerCocoa.h:
* Source/WebCore/platform/graphics/ca/cocoa/PlatformCALayerCocoa.mm:
(WebCore::PlatformCALayerCocoa::PlatformCALayerCocoa):
(WebCore::PlatformCALayerCocoa::clone const):
(WebCore::PlatformCALayerCocoa::appleVisualEffect const):
(WebCore::PlatformCALayerCocoa::setAppleVisualEffect):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::repaintBeforeStyleChange):
(WebCore::RenderElement::adjustFragmentedFlowStateOnContainingBlockChangeIfNeeded):
* Source/WebCore/rendering/RenderElement.h:
* Source/WebCore/rendering/RenderElementInlines.h:
(WebCore::RenderElement::hasAppleVisualEffect const):
(WebCore::RenderElement::hasAppleVisualEffectRequiringBackdropFilter const):
(WebCore::RenderElement::canContainAbsolutelyPositionedObjects const):
(WebCore::RenderElement::canContainFixedPositionObjects const):
(WebCore::RenderElement::createsGroupForStyle):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::canCreateStackingContext):
(WebCore::RenderLayer::computeCanBeBackdropRoot const):
(WebCore::RenderLayer::debugDescription const):
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::createPrimaryGraphicsLayer):
(WebCore::RenderLayerBacking::updateBackdropFiltersGeometry):
(WebCore::RenderLayerBacking::updateAppleVisualEffect):
(WebCore::RenderLayerBacking::updateConfigurationAfterStyleChange):
(WebCore::RenderLayerBacking::updateGeometry):
* Source/WebCore/rendering/RenderLayerBacking.h:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::CompositingState::updateWithDescendantStateAndLayer):
(WebCore::RenderLayerCompositor::requiresOwnBackingStore const):
(WebCore::RenderLayerCompositor::reasonsForCompositing const):
(WebCore::RenderLayerCompositor::requiresCompositingForFilters const):
* Source/WebCore/rendering/RenderLayerInlines.h:
(WebCore::RenderLayer::hasAppleVisualEffect const):
(WebCore::RenderLayer::hasAppleVisualEffectRequiringBackdropFilter const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::rareDataChangeRequiresLayout):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::hasAppleVisualEffect const):
(WebCore::RenderStyle::hasAppleVisualEffectRequiringBackdropFilter const):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/LayerProperties.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::drawInContext):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm:
(WebKit::materialRecipeForAppleVisualEffect):
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToLayer):

Set a &quot;recipe&quot; on `MTMaterialLayer` to achieve the blur effect.

* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(WebKit::dumpChangedLayers):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::makeNode):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeHostIOS.mm:
(WebKit::RemoteLayerTreeHost::makeNode):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeViews.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeViews.mm:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.mm:
(WebKit::PlatformCALayerRemote::appleVisualEffect const):
(WebKit::PlatformCALayerRemote::setAppleVisualEffect):

Canonical link: <a href="https://commits.webkit.org/288125@main">https://commits.webkit.org/288125@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f6162a8b5030b4f258908b4bb486e2218fb2420b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81878 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1404 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35832 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86435 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32881 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/83984 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1437 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9227 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63863 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21589 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84948 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1063 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74514 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44147 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/963 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28691 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31327 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72311 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29300 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87866 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9118 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6499 "Found 1 new test failure: http/wpt/mediarecorder/pause-recording-timeSlice.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72235 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9303 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70333 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71450 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15538 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14463 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/517 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12698 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9069 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14607 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8910 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12433 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10718 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/platform/graphics/GraphicsLayer.h:
```diff
@@ -849,6 +858,10 @@ class GraphicsLayer : public RefCounted<GraphicsLayer> {
     std::optional<FloatRect> m_animationExtent;
 
+#if HAVE(CORE_MATERIAL)
+    AppleVisualEffect m_appleVisualEffect { AppleVisualEffect::None };
```
Could you find an 8-bit member to group this with to optimize padding?

pxlcoder replied:
Moved in between two members where there was 1 byte of padding.

smfr commented on Source/WebCore/rendering/RenderLayerBacking.cpp:
```diff
@@ -782,8 +785,12 @@ void RenderLayerBacking::updateBackdropFilters(const RenderStyle& style)
-        return;
+    if (!m_canCompositeBackdropFilters) {
+#if HAVE(CORE_MATERIAL)
+        if (!renderer().hasAppleVisualEffectRequiringBackdropFilter())
```
Not really a fan of control flow inside #Ifdefs. Please rejigger.

pxlcoder committed:
[Materials] Add rendering support for blur effects
https://bugs.webkit.org/show_bug.cgi?id=284729
rdar://141527633

Reviewed by Simon Fraser.

Create and display `MTMaterialLayer`s when using `-apple-visual-effect` with
one of the `-apple-system-blur-` keywords. Note that this is not web-exposed.

`MTMaterialLayer`s are backdrop layers with a specific set of `CAFilter`s,
depending on the blur effect.

* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-chrome-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-chrome.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thick-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thick.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thin-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-thin.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-ultra-thin-expected.txt: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material-ultra-thin.html: Added.
* LayoutTests/apple-visual-effects/apple-visual-effect-blur-material.html: Added.
* LayoutTests/apple-visual-effects/backdrop-filter-with-apple-visual-effect-expected.html: Added.
* LayoutTests/apple-visual-effects/backdrop-filter-with-apple-visual-effect.html: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-chrome-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-thick-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-thin-expected.txt: Added.
* LayoutTests/platform/ios/apple-visual-effects/apple-visual-effect-blur-material-ultra-thin-expected.txt: Added.
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/PlatformMac.cmake:
* Source/WebCore/PAL/pal/cocoa/CoreMaterialSoftLink.h: Copied from Source/WebCore/platform/cocoa/AppleVisualEffect.h.
* Source/WebCore/PAL/pal/cocoa/CoreMaterialSoftLink.mm: Copied from Source/WebCore/platform/cocoa/AppleVisualEffect.h.
* Source/WebCore/PAL/pal/spi/cocoa/CoreMaterialSPI.h: Copied from Source/WebCore/platform/cocoa/AppleVisualEffect.h.
* Source/WebCore/page/FrameSnapshotting.cpp:
(WebCore::styleContainsComplexBackground):
* Source/WebCore/platform/cocoa/AppleVisualEffect.cpp:
(WebCore::appleVisualEffectNeedsBackdrop):

Add a helper method to distinguish visual effects which create a backdrop (blur)
from those that do not (vibrancy). Support for vibrancy will be added in a
subsequent patch.

* Source/WebCore/platform/cocoa/AppleVisualEffect.h:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::needsBackdrop const):
(WebCore::GraphicsLayer::dumpProperties const):
* Source/WebCore/platform/graphics/GraphicsLayer.h:
(WebCore::GraphicsLayer::appleVisualEffect const):
(WebCore::GraphicsLayer::setAppleVisualEffect):

Specify `BackdropFiltersChanged` in addition to `AppleVisualEffectChanged`, as
material blurs set filters on a backdrop. A structural layer must be created,
and the backdrop filter rect must also be updated.

(WebCore::GraphicsLayer::needsBackdrop const): Deleted.
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::setAppleVisualEffect):
(WebCore::GraphicsLayerCA::commitLayerChangesBeforeSublayers):
(WebCore::GraphicsLayerCA::updateBackdropFilters):
(WebCore::GraphicsLayerCA::updateAppleVisualEffect):

Only support backdrop effects for now. Foreground effects (vibrancy) will be
implemented in a subsequent patch.

(WebCore::GraphicsLayerCA::purposeNameForInnerLayer const):
(WebCore::GraphicsLayerCA::layerChangeAsString):
(WebCore::GraphicsLayerCA::changeLayerTypeTo):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.h:
* Source/WebCore/platform/graphics/ca/PlatformCALayer.h:

Introduce a new layer type for materials.

* Source/WebCore/platform/graphics/ca/PlatformCALayer.mm:
(WebCore::operator<<):
* Source/WebCore/platform/graphics/ca/cocoa/PlatformCALayerCocoa.h:
* Source/WebCore/platform/graphics/ca/cocoa/PlatformCALayerCocoa.mm:
(WebCore::PlatformCALayerCocoa::PlatformCALayerCocoa):
(WebCore::PlatformCALayerCocoa::clone const):
(WebCore::PlatformCALayerCocoa::appleVisualEffect const):
(WebCore::PlatformCALayerCocoa::setAppleVisualEffect):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::repaintBeforeStyleChange):
(WebCore::RenderElement::adjustFragmentedFlowStateOnContainingBlockChangeIfNeeded):
* Source/WebCore/rendering/RenderElement.h:
* Source/WebCore/rendering/RenderElementInlines.h:
(WebCore::RenderElement::hasAppleVisualEffect const):
(WebCore::RenderElement::hasAppleVisualEffectRequiringBackdropFilter const):
(WebCore::RenderElement::canContainAbsolutelyPositionedObjects const):
(WebCore::RenderElement::canContainFixedPositionObjects const):
(WebCore::RenderElement::createsGroupForStyle):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::canCreateStackingContext):
(WebCore::RenderLayer::computeCanBeBackdropRoot const):
(WebCore::RenderLayer::debugDescription const):
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::createPrimaryGraphicsLayer):
(WebCore::RenderLayerBacking::updateBackdropFiltersGeometry):
(WebCore::RenderLayerBacking::updateAppleVisualEffect):
(WebCore::RenderLayerBacking::updateConfigurationAfterStyleChange):
(WebCore::RenderLayerBacking::updateGeometry):
* Source/WebCore/rendering/RenderLayerBacking.h:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::CompositingState::updateWithDescendantStateAndLayer):
(WebCore::RenderLayerCompositor::requiresOwnBackingStore const):
(WebCore::RenderLayerCompositor::reasonsForCompositing const):
(WebCore::RenderLayerCompositor::requiresCompositingForFilters const):
* Source/WebCore/rendering/RenderLayerInlines.h:
(WebCore::RenderLayer::hasAppleVisualEffect const):
(WebCore::RenderLayer::hasAppleVisualEffectRequiringBackdropFilter const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::rareDataChangeRequiresLayout):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::hasAppleVisualEffect const):
(WebCore::RenderStyle::hasAppleVisualEffectRequiringBackdropFilter const):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/LayerProperties.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::drawInContext):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm:
(WebKit::materialRecipeForAppleVisualEffect):
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToLayer):

Set a "recipe" on `MTMaterialLayer` to achieve the blur effect.

* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(WebKit::dumpChangedLayers):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::makeNode):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeHostIOS.mm:
(WebKit::RemoteLayerTreeHost::makeNode):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeViews.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeViews.mm:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.mm:
(WebKit::PlatformCALayerRemote::appleVisualEffect const):
(WebKit::PlatformCALayerRemote::setAppleVisualEffect):

Canonical link: https://commits.webkit.org/288125@main
